iso-authoryear.bbx

```diff
@@ -2,23 +2,29 @@
 
 \RequireBibliographyStyle{iso}
 \ExecuteBibliographyOptions{%
-  sorting=nyt
+  sorting=nyt,
 }
 
-\def\printextrayear{%
-  \ifentrytype{patent}
-    {}
-    {\printfield{year}\printfield{extradate}}%
+% The field extradate turns from integer type to a letter
+% after year to distinguish more works of the author
+% from the same year, e.g. 1994a, 1994b.
+% If there is no date field in the bib entry,
+% we should state there is no date by printing
+% a localisation string, e.g. [n.d.].
+\newcommand{\printextrayear}{%
+  \iffieldundef{year}
+    {\bibstring[\mkbibbrackets]{nodate}}%
+    {\printdateextra}%
```

Comment:
This introduces a regression for printing full date by `\mkdaterangeiso{}`.

Reply:
What kind of regression? Do you have an MWE (my holidays just ended, so I'm shorter on time)?

Reply:
Sure, I was not so precise in the comment.. here is an MWE:
```
\documentclass{article}
\usepackage{filecontents}
\begin{filecontents}{example.bib}
@patent{Groll2008,
  author      = {Clad Metals LLC Canonsburg, PA 15317 (US)},
  title       = {Method of making a copper core five-ply composite
                   and cooking vessel},
  editora     = {Groll, W. A.},
  editoratype = {inventor},
  publisher   = {Google Patents},
  number      = {EP 1 094 937 B1},
  type        = {patenteu},
  date        = {2008-07-30},
  url         = {https://patents.google.com/patent/EP1094937B1},
}
\end{filecontents}

\usepackage[style=iso-authoryear,backend=biber]{biblatex}
\addbibresource{example.bib}

\begin{document}
\nocite{*}
\printbibliography
\end{document}
```
The output is (date is not in format YYYY-MM-DD):
![Screenshot_2020-01-06 mwe1-anotation pdf](https://user-images.githubusercontent.com/11637888/71854106-8249c600-30dd-11ea-8462-e308be6d44db.png)

For `style=iso-numeric`, the output is (date is in correct format YYYY-MM-DD):
![Screenshot_2020-01-06 mwe2-annotation pdf](https://user-images.githubusercontent.com/11637888/71854234-d3f25080-30dd-11ea-81d0-82b7568ebf23.png)

Commands `\printdate` and/or `\printdateextra` somehow interfere the command `\mkdaterangeiso{}`.

Reply:
Ah, I see. Thank your for the MWE. This was a bug in `biblatex`: In certain situations `\printdate` would delete fields like `month` and `day`. The idea was that that would only be locally, but due to missing grouping the fields might be cleared globally. This is fixed in https://github.com/plk/biblatex/commit/34952445cc2141f0f071ae5b6c6018ac6b4b3453. I will try to push a fix to this branch.

Reply:
Thanks a lot for the fix!